src/utils/promisify.js

```diff
@@ -0,0 +1,21 @@
+export const stat = (inputFileSystem, path) => {
+    return new Promise((resolve, reject) => {
+        inputFileSystem.stat(path, (err, stats) => {
+            if (err) {
+                reject(err);
+            }
```

Comment:
Without an `else` or early `return` statement, `resolve` will even be called in the case of a rejection. Same is the case for `readFile` below.

Reply:
@ronkorving reject already do early exit, no necessary `return`

Reply:
How can it? To my knowledge, the only way to return early in JS is `return` or `throw`.
`reject(err)` is just a function call.

Reply:
@ronkorving `reject()` === `throw` inside promise

Reply:
It is true that `resolve(stats)` will still be called, but the `stats` value will not be passed on to anything. Even if there is another promise waiting for the resolve value, the rejection path will be taken instead.

It's a still a bug though, in the sense that it was never intended for the `resolve` to run in the first place.

@evilebottnawi WDYT of me doing another PR that adds back `pify` with the alternative syntax that works on ES6 classes?

Reply:
@filipesilva if you have time for this, it is not critical problem :+1:

Reply:
@ronkorving IMHO, `return` here need only for easy reading and understanding code. Feel free to PR if this confusing for you

Reply:
@evilebottnawi I don't think I'm confused. As @filipesilva also pointed out, `resolve(stats)` *will* still be called.

I'll leave it to you both to figure out the way forward, given that there seems to be a desire to put `pify` back in anyway.

Reply:
I tried https://github.com/webpack-contrib/copy-webpack-plugin/pull/228 but just using `pify` on the single method doesn't seem to work well either, `this` seems to be undefined when doing that so class members aren't accessible.

@ronkorving would you like to make a PR to change the `if` to a `if ... else`?

Reply:
```js
if (err) reject(err) // I'm not 100% here, but this should be === (x) => x (return)

resolve(x)
```

```js
if (err) {
   return reject(err)
}

resolve (x)
```

```js
if (err) {
   reject(err)
   // eventually to guard against 'runaway' promises
   return null
}

resolve(x)
```

Please no extra lib, it's not necessary to `promisify`  two callback functions :)

Reply:
@ronkorving Thx. Nice catch :)